crates/storage/provider/src/providers/database/metrics.rs

```diff
@@ -0,0 +1,66 @@
+use metrics::Histogram;
+use reth_metrics::Metrics;
+use std::time::{Duration, Instant};
+
+#[derive(Debug)]
+pub(crate) struct DurationsRecorder {
+    pub(crate) actions: Vec<(Action, Duration)>,
+    latest: Instant,
+}
+
+impl Default for DurationsRecorder {
+    fn default() -> Self {
+        Self { actions: Vec::new(), latest: Instant::now() }
+    }
+}
+
+impl DurationsRecorder {
+    /// Saves the provided duration for future logging and instantly reports as a metric with
+    /// `action` label.
+    pub(crate) fn record_duration(&mut self, action: Action, duration: Duration) {
+        self.actions.push((action, duration));
+        Metrics::new_with_labels(&[("action", format!("{action:?}"))]).duration.record(duration);
```

Comment:
allocating a string here seems unnecessary if we can use static &str identifiers for the Action, I believe this call should work for `&'static str`

Reply:
agree